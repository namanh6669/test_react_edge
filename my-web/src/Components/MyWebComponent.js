
import React from 'react';
// eslint-disable-next-line
import MyTabs from './MyTabs.js'
import MyTables from './MyTables.js'
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import FlatButton from 'material-ui/FlatButton';
function handleTouchTap() {
    alert('Back To Homepage !!!');
}

const styles = {
    title: {
        cursor: 'pointer',

    },
    appbar: {
        float: 'right',
        width: '40%',

    },
    appbar1: {
        float: 'left',
        width: '100%',
    },
    appbar2: {
        float: 'right',
    },

};


class MyWebComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            value: 'Decrease',
        };
    }
    callbackTypeSort(value) {
       this.setState({
           value: value
       }) 
    }

    render() {
        console.log(this.state.value)
        return (
            <div>
                <AppBar
                    title={<span style={styles.title}>S&P/ASX</span>}
                    onTitleTouchTap={handleTouchTap}
                    showMenuIconButton={false}>
                    <MyTabs callbackTypeSort={(value)=>this.callbackTypeSort(value)}></MyTabs>
                </AppBar>
                <div>
                    <MyTables Type = {this.state.value}></MyTables>
                </div>
            </div>
        )
    }
 
}
export default MyWebComponent;
