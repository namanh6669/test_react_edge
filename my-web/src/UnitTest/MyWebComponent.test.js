import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import MyWebComponent from '../Components/MyWebComponent'

test('Render MyWebComponent',()=>{
    const componets = shallow(<MyWebComponent/>);
    expect(toJson(componets)).toMatchSnapshot();
})