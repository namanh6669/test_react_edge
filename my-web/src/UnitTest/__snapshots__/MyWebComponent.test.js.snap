// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyWebComponent 1`] = `
<div>
  <AppBar
    onTitleTouchTap={[Function]}
    showMenuIconButton={false}
    title={
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        S&P/ASX
      </span>
    }
    zDepth={1}
  >
    <MyTabs
      callbackTypeSort={[Function]}
    />
  </AppBar>
  <div>
    <MyTables
      Type="Increase"
    />
  </div>
</div>
`;
